packaging/gap2seq-recipe/build.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
```

Comment:
Why is this recipe living here instead of in bioconda?

Reply:
@tomkinsc can you review this part?

Reply:
Because I was making a custom version of it, just as we have a custom version of kraken.  Since I'm not the gap2seq author I don't feel comfortable distributing unofficial/modified versions of it on bioconda, but for our use on the broad-viral channel it should be ok.

Reply:
Oh! I didn't realize that. Why are we maintaining a custom fork of it? For Kraken it was an absolutely last resort, and generally we avoid it and prefer to either get important changes merged back into the main repo or write workarounds in our wrappers. In this particular case, [I don't see](https://github.com/broadinstitute/Gap2Seq/network) any new code that we have that the original repo doesn't... what does our version have?

Reply:
It was actually work-in-progress to localize reads to gaps; I can move that to a future PR.

Reply:
Yeah, I’d prefer to keep `packaging/` in this repo specifically for viral-ngs-related packaging, with a seperate repo for the gap2seq fork. It’s probably OK to include it in bioconda if the notes are explicit about what is different in your fork, and the recipe links to the fork’s repo so people can see the commit log with relevant changes. If the gap2seq’s license allows remixes, it’s prefectly OK to distributed a modified version as long as the original source is attributed how it wants to be and there’s transparency about the changes. A PR to gap2seq would be best, but if we have to fork it we have to fork it...

Reply:
It sounds like the current version of this particular PR would work with the off-the-shelf gap2seq, in which case maybe we should be using the bioconda package instead of the broad-viral one.

Also, some of your localizing reads question is a concern about an edge case that we're not sure how often it causes a problem, and if it does, I bet a lot of that might be alleviated simply by passing in taxfilt.bam instead of cleaned.bam (though I'd prefer not to do that unless we know it's a problem that happens often).

Reply:
I've tried passing in taxfilt.bam but that leads to fewer gaps being closed.  It's one of the parameters of being conservative vs closing more gaps, another being the minimum coverage of a kmer for it to be considered "solid".  I can make these parameters that can be set in config.yaml .